### PR TITLE
Adding PinterestApi and PinterestExample

### PR DESCRIPTION
**Pull request by martinmalek**
Support for Pinterest Api

### CODE REVIEW

**Comment by martinmalek** [comment]
Let me know if you want me to send you app id and secret for a quick test.

File: scribejava-apis/src/test/java/com/github/scribejava/apis/examples/PinterestExample.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package com.github.scribejava.apis.examples;
+
+import com.github.scribejava.apis.PinterestApi;
+import com.github.scribejava.core.builder.ServiceBuilder;
+import com.github.scribejava.core.model.*;
```
Comment (kullfar) [inline scribejava-apis/src/test/java/com/github/scribejava/apis/examples/PinterestExample.java]: Not so critical, but it's better not to use star imports.

File: scribejava-apis/src/main/java/com/github/scribejava/apis/PinterestAPI.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import com.github.scribejava.core.builder.api.DefaultApi20;
+import com.github.scribejava.core.extractors.AccessTokenExtractor;
+import com.github.scribejava.core.extractors.JsonTokenExtractor;
+import com.github.scribejava.core.model.OAuthConfig;
+import com.github.scribejava.core.model.OAuthConstants;
+import com.github.scribejava.core.model.Verb;
+import com.github.scribejava.core.utils.OAuthEncoder;
+import com.github.scribejava.core.utils.Preconditions;
+
+public class PinterestApi extends DefaultApi20 {
```
Comment (kullfar) [inline scribejava-apis/src/main/java/com/github/scribejava/apis/PinterestAPI.java]: non compilable class
it's name 'PinterestApi'
while file name is 'PinterestAPI'

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.3:compile (default-compile) on project scribejava-apis: Compilation failure
[ERROR] /tmp/scribejava/scribejava-apis/src/main/java/com/github/scribejava/apis/PinterestAPI.java:[12,8] class PinterestApi is public, should be declared in a file named PinterestApi.java


looks like you use Windows or some case-insensitive FS

**Comment by kullfar** [comment]
Thanks for Pull request!
And yes, provide test credentials to email mentioned here please
https://github.com/scribejava/scribejava/blob/master/pom.xml#L48

File: scribejava-apis/src/main/java/com/github/scribejava/apis/PinterestAPI.java (hunk 2 above)
Comment (martinmalek) [inline scribejava-apis/src/main/java/com/github/scribejava/apis/PinterestAPI.java]: Updated

Comment (martinmalek) [inline scribejava-apis/src/main/java/com/github/scribejava/apis/PinterestAPI.java]: Fixed. Please try again.

Martin Malek
Product Manager - Channels, API & Integrations
Copenhagen
Falcon Social
H.C. Andersens Blvd. 27
1553 Copenhagen
Budapest
Falcon Social
Colabs Startup Center Zrt
1016 Budapest, Krisztina krt. 99
 <https://www.falconsocial.com/?utm_source=Employee%20emails&utm_medium=emai
l&utm_content=Martin%20Malek&utm_campaign=Mail%20signature>
Social Media Management for Enterprise

From:  Stas Gromov <notifications@github.com>
Reply-To:  scribejava/scribejava
<reply+00095714c419c3992da322e3ba1638ecf8bfa06c5155eb4b92cf00000001129925149
2a170ce02e3e20f@reply.github.com>
Date:  Monday 28 December 2015 at 17:28
To:  scribejava/scribejava <scribejava@noreply.github.com>
Cc:  Martin Malek <martin.malek@falconsocial.com>
Subject:  Re: [scribejava] Adding PinterestApi and PinterestExample (#591)


In 
scribejava-apis/src/main/java/com/github/scribejava/apis/PinterestAPI.java
<https://github.com/scribejava/scribejava/pull/591#discussion_r48488975> :
> @@ -0,0 +1,44 @@
> +package com.github.scribejava.apis;
> +
> +import com.github.scribejava.core.builder.api.DefaultApi20;
> +import com.github.scribejava.core.extractors.AccessTokenExtractor;
> +import com.github.scribejava.core.extractors.JsonTokenExtractor;
> +import com.github.scribejava.core.model.OAuthConfig;
> +import com.github.scribejava.core.model.OAuthConstants;
> +import com.github.scribejava.core.model.Verb;
> +import com.github.scribejava.core.utils.OAuthEncoder;
> +import com.github.scribejava.core.utils.Preconditions;
> +
> +public class PinterestApi extends DefaultApi20 {
non compilable class
it's name 'PinterestApi'
while file name is 'PinterestAPI'

[ERROR] Failed to execute goal
org.apache.maven.plugins:maven-compiler-plugin:3.3:compile (default-compile)
on project scribejava-apis: Compilation failure
[ERROR] 
/tmp/scribejava/scribejava-apis/src/main/java/com/github/scribejava/apis/Pin
terestAPI.java:[12,8] class PinterestApi is public, should be declared in a
file named PinterestApi.java

looks like you use Windows or some case-insensitive FS


Reply to this email directly or view it on GitHub
<https://github.com/scribejava/scribejava/pull/591/files#r48488975> .

**Comment by martinmalek** [comment]
Comment changes applied. Please try again.

**Comment by kullfar** [comment]
The code looks good. I'm trying to run the example and will merge after that.

**Pull request merged**